src/data/finset/preimage.lean

```diff
@@ -34,6 +34,38 @@ set.finite.mem_to_finset
   (hf : set.inj_on f (f ⁻¹' ↑s)) : (↑(preimage s f hf) : set α) = f ⁻¹' ↑s :=
 set.finite.coe_to_finset _
 
+@[simp] lemma preimage_empty {f : α → β} : preimage ∅ f (by simp [inj_on]) = ∅ :=
+finset.coe_injective (by simp)
+
+@[simp] lemma preimage_univ {f : α → β} [fintype α] [fintype β] (hf) :
+  preimage univ f hf = univ :=
+finset.coe_injective (by simp)
+
+@[simp] lemma preimage_inter [decidable_eq α] [decidable_eq β] {f : α → β} {s t : finset β}
+  (hs : set.inj_on f (f ⁻¹' ↑s)) (ht : set.inj_on f (f ⁻¹' ↑t)) :
+  preimage (s ∩ t) f (λ x₁ hx₁ x₂ hx₂ hf, by {
+    simp at hx₁ hx₂,
+    exact hs hx₁.1 hx₂.1 hf,
+  }) = preimage s f hs ∩ preimage t f ht :=
+finset.coe_injective (by simp)
```

Comment:
Note that this one is a conditional simp lemma, as `hs` and `ht` cannot be inferred from the LHS.